package/README.md

```diff
@@ -563,9 +563,23 @@ class MyApp extends StatelessWidget {
   - `BeamLocation` keeps query and path parameters from URI in its `BeamState`. The `:` is necessary in `pathPatterns` if you _might_ get path parameter from browser.
 
 - `BeamPage`'s child is an arbitrary `Widget` that represents your app screen / page.
-  - `key` is important for `Navigator` to optimize rebuilds. This should be a unique value for "page state".
+  - `key` is important for `Navigator` to optimize rebuilds. This needs to be a unique value (e.g. [ValueKey](https://api.flutter.dev/flutter/foundation/ValueKey-class.html)) for "page state". (see [Keys](#keys))
   - `BeamPage` creates `MaterialPageRoute` by default, but other transitions can be chosen by setting `BeamPage.type` to one of available `BeamPageType`.
 
+## Keys
+
+When you beam somewhere, you are basically putting a new list of "pages" into `Navigator.pages`.
+
+Now the Navigator has to decide on the transition between the old list of pages and the new list of pages.
+
+In order to know which pages changed and which pages stayed the same, Navigator looks at the pages `key`s. If the page of 2 pages that are compared are equals (important here: `null` == `null`), Navigator treats them as the same page and does not issue a rebuild.
+
+So, `key` is a pretty important field on the `BeamPage`s. You should always provide one (most likely a [ValueKey](https://api.flutter.dev/flutter/foundation/ValueKey-class.html) will be good).
+
+If you do not provide `key`s to your `BeamPage`s, the behavior you will see is this:
+
+You beam to some location with `Beamer.of(context).beamToNamed('/somewhere')`, you will see no change in your UI. The new BeamPage won't get pushed (since Navigator thinks it is the same as the already displayed one).
```

Comment:
I think it would look nicer to write this as:

"You beam to some location with `Beamer.of(context).beamToNamed('/somewhere')`, but see no change in your UI. The new `BeamPage` doesn't get pushed since `Navigator` thinks it is the same as the already displayed one."

Reply:
Agreed, done